src/charm.py

```diff
@@ -189,11 +189,14 @@ def _generate_root_certificate(self) -> None:
         if self._root_certificate_is_stored:
             secret = self.model.get_secret(label=CA_CERTIFICATES_SECRET_LABEL)
             secret.set_content(content=secret_content)
+            secret.set_info(
+                expire=timedelta(days=self._config_root_ca_certificate_validity),
+            )
```

Comment:
It seems to me that this branch is currently [impossible to reach](https://github.com/canonical/self-signed-certificates-operator/blob/721df10854666ab6a73fa1b46a89ee3ba791e607/src/charm.py#L216).